src/Microsoft.DotNet.ProjectModel/WorkspaceContext.cs

```diff
@@ -128,31 +145,93 @@ public void Refresh()
+        public ProjectContext GetProjectContext(string projectPath, NuGetFramework framework) => GetProjectContext(projectPath, framework, EmptyArray<string>.Value);
+        public ProjectContext GetProjectContext(string projectPath, NuGetFramework framework, IEnumerable<string> runtimeIdentifier)
         {
-            return GetProjectContextCollection(projectPath).ProjectContexts;
+            var contexts = GetProjectContextCollection(projectPath);
+            if (contexts == null)
+            {
+                return null;
+            }
+
+            return contexts
+                .ProjectContexts
+                .FirstOrDefault(c => Equals(c.TargetFramework, framework) && RidsMatch(c.RuntimeIdentifier, runtimeIdentifier));
+        }
+
+        public ProjectContext GetRuntimeContext(ProjectContext context, IEnumerable<string> runtimeIdentifiers)
+        {
+            var contexts = GetProjectContextCollection(context.ProjectDirectory);
+            if (contexts == null)
+            {
+                return null;
+            }
+
+            var runtimeContext = runtimeIdentifiers
+                .Select(r => contexts.GetTarget(context.TargetFramework, r))
+                .FirstOrDefault(c => c != null);
+
+            if (runtimeContext == null)
+            {
+                if (context.IsPortable)
+                {
+                    // We're specializing a portable target, so synthesize a runtime target manually
+                    // We don't cache this project context, but we'll still use the cached Project and LockFile
+                    return InitializeProjectContextBuilder(context.ProjectFile)
+                        .WithTargetFramework(context.TargetFramework)
+                        .WithRuntimeIdentifiers(runtimeIdentifiers)
+                        .Build();
+                }
+
+                // We are standalone, but don't support this runtime
+                var rids = string.Join(", ", runtimeIdentifiers);
+                throw new InvalidOperationException($"Can not find runtime target for framework '{context.TargetFramework}' compatible with one of the target runtimes: '{rids}'. " +
+                                                    "Possible causes:" + Environment.NewLine +
+                                                    "1. The project has not been restored or restore failed - run `dotnet restore`" + Environment.NewLine +
+                                                    $"2. The project does not list one of '{rids}' in the 'runtimes' section.");
+            }
+
+            return runtimeContext;
         }
 
+        public Project GetProject(string projectDirectory) => GetProjectCore(projectDirectory)?.Model;
+
         public ProjectContextCollection GetProjectContextCollection(string projectPath)
         {
+            var normalizedPath = NormalizeProjectPath(projectPath);
+            if (normalizedPath == null)
+            {
+                return null;
```

Comment:
I'm not really sure about this. Returning `null` feels like a new contract. Doesn't it break any client?